File: urbanairship-layout/src/testDebug/java/com/urbanairship/android/layout/util/PercentUtilsTest.java
package com.urbanairship.android.layout.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class PercentUtilsTest {

    @Test
    public void parsesPercentString() {
        assertEquals(0f, PercentUtils.parse("0%"), 0.0001);
        assertEquals(0f, PercentUtils.parse("0.0%"), 0.0001);
        assertEquals(0f, PercentUtils.parse("0.00%"), 0.0001);

        assertEquals(0.5f, PercentUtils.parse("50%"), 0.0001);
        assertEquals(0.5f, PercentUtils.parse("50.0%"), 0.0001);
        assertEquals(0.5f, PercentUtils.parse("50.00%"), 0.0001);

        assertEquals(1f, PercentUtils.parse("100%"), 0.0001);
        assertEquals(1f, PercentUtils.parse("100.0%"), 0.0001);
        assertEquals(1f, PercentUtils.parse("100.00%"), 0.0001);
    }

    @Test
    public void parsesPercentStringWithoutSymbol() {
        assertEquals(0f, PercentUtils.parse("0"), 0.0001);
        assertEquals(0.5f, PercentUtils.parse("50"), 0.0001);
        assertEquals(1f, PercentUtils.parse("100"), 0.0001);
    }

    @Test
    public void isPercentMatches() {
        assertTrue(PercentUtils.isPercent("0%"));
        assertTrue(PercentUtils.isPercent("50%"));
        assertTrue(PercentUtils.isPercent("100%"));

        assertFalse(PercentUtils.isPercent("dog"));
        assertFalse(PercentUtils.isPercent("9000%"));
        assertFalse(PercentUtils.isPercent("10.25%"));
        assertFalse(PercentUtils.isPercent("something 10%"));
    }
}

File: urbanairship-layout/src/test/java/com/urbanairship/android/layout/util/PercentUtilsTest.kt
package com.urbanairship.android.layout.util

import org.junit.Assert
import org.junit.Test

class PercentUtilsTest {

    @Test
    fun parsesPercentString() {
        Assert.assertEquals(0.0, PercentUtils.parse("0%").toDouble(), 0.0001)
        Assert.assertEquals(0.0, PercentUtils.parse("0.0%").toDouble(), 0.0001)
        Assert.assertEquals(0.0, PercentUtils.parse("0.00%").toDouble(), 0.0001)

        Assert.assertEquals(0.5, PercentUtils.parse("50%").toDouble(), 0.0001)
        Assert.assertEquals(0.5, PercentUtils.parse("50.0%").toDouble(), 0.0001)
        Assert.assertEquals(0.5, PercentUtils.parse("50.00%").toDouble(), 0.0001)

        Assert.assertEquals(1.0, PercentUtils.parse("100%").toDouble(), 0.0001)
        Assert.assertEquals(1.0, PercentUtils.parse("100.0%").toDouble(), 0.0001)
        Assert.assertEquals(1.0, PercentUtils.parse("100.00%").toDouble(), 0.0001)
    }

    @Test
    fun parsesPercentStringWithoutSymbol() {
        Assert.assertEquals(0.0, PercentUtils.parse("0").toDouble(), 0.0001)
        Assert.assertEquals(0.5, PercentUtils.parse("50").toDouble(), 0.0001)
        Assert.assertEquals(1.0, PercentUtils.parse("100").toDouble(), 0.0001)
    }

    @Test
    fun isPercentMatches() {
        Assert.assertTrue(PercentUtils.isPercent("0%"))
        Assert.assertTrue(PercentUtils.isPercent("50%"))
        Assert.assertTrue(PercentUtils.isPercent("100%"))

        Assert.assertFalse(PercentUtils.isPercent("dog"))
        Assert.assertFalse(PercentUtils.isPercent("9000%"))
        Assert.assertFalse(PercentUtils.isPercent("10.25%"))
        Assert.assertFalse(PercentUtils.isPercent("something 10%"))
    }
}
